realm/realm-library/src/objectServer/java/io/realm/internal/objectserver/ObjectServerSession.java

```diff
@@ -108,6 +113,8 @@
     private SyncSession userSession;
     private SyncSession publicSession;
 
+    private final static ScheduledThreadPoolExecutor REFRESH_TOKENS_EXECUTOR = new ScheduledThreadPoolExecutor(1);
```

Comment:
Good point. Multiple users are supported though, but you are right that e.g `currentUsers()` will crash for multiple logged in users, so perhaps starting out with 1 thread is fine. Then we can adjust if need be.